src/abstract-interpretation/handler/binop/binop.ts

```diff
@@ -0,0 +1,45 @@
+import {Handler} from '../handler'
+import {AINode} from '../../processor'
+import {BinaryOperatorFlavor, ParentInformation, RBinaryOp} from '../../../r-bridge'
+import {guard} from '../../../util/assert'
+import {operators} from './operators'
+
+export type BinOpOperators = {
+	[key in BinaryOperatorFlavor]: (lhs: AINode, rhs: AINode, node: RBinaryOp<ParentInformation>) => AINode
+}
+
+export class BinOp implements Handler<AINode> {
+	lhs:           AINode | undefined
+	rhs:           AINode | undefined
+	readonly node: RBinaryOp<ParentInformation>
+
+	constructor(node: RBinaryOp<ParentInformation>) {
+		this.node = node
+	}
+
+	getName(): string {
+		return `Bin Op (${this.node.flavor})`
+	}
+
+	enter(): void {
+		console.log(`Entered ${this.getName()}`)
+	}
+
+	exit(): AINode {
+		console.log(`Exited ${this.getName()}`)
+		guard(this.lhs !== undefined, `No LHS found for assignment ${this.node.info.id}`)
+		guard(this.rhs !== undefined, `No RHS found for assignment ${this.node.info.id}`)
+		return operators[this.node.flavor](this.lhs, this.rhs, this.node)
```

Comment:
Maybe we should abstract over the operators as well. If we want to do, say, type inference nothing from BinOp has to change besides the operator processing (and the generic but this is a type-level thing). So we could reuse these components.
Is a factory good for that? i am not too sure, but it may be sufficient to assign that statically for now (but outside of the BinOp scope).

Reply:
Tracked in #608